content/develop/connect/clients/client-side-caching.md

```diff
@@ -0,0 +1,187 @@
+    [`BITFIELD`]({{< relref "/commands/bitfield" >}}) on a string
+    caches the returned values separately from the original string.
+-   For composite data types accessed by keys
+    ([hash]({{< relref "/develop/data-types/hashes" >}}),
+    [JSON]({{< relref "/develop/data-types/json" >}}),
+    [set]({{< relref "/develop/data-types/sets" >}}), and
+    [sorted set]({{< relref "/develop/data-types/sorted-sets" >}})),
+    the whole object is cached separately from the individual fields.
+    So the results of `HGETALL mykey` and `HGET mykey myfield` create
+    separate entries in the cache.
+-   Ranges from [lists]({{< relref "/develop/data-types/lists" >}}),
+    [streams]({{< relref "/develop/data-types/streams" >}}),
+    and [sorted sets]({{< relref "/develop/data-types/sorted-sets" >}})
+    are cached separately from the object they form a part of. Likewise,
+    subsets returned by [`SINTER`]({{< relref "/commands/sinter" >}}) and
+    [`SDIFF`]({{< relref "/commands/sdiff" >}}) create separate cache entries.
+-   For multi-key read commands such as [`MGET`]({{< relref "/commands/mget" >}}),
+    the ordering of the keys is significant. For example `MGET name:1 name:2` is
+    cached separately from `MGET name:2 name:1` because the server returns the
+    values in the order you specify.
+-   Boolean or numeric values calculated from data types (for example 
+    [`SISMEMBER`]({{< relref "/commands/sismember" >}})) and
+    [`LLEN`]({{< relref "/commands/llen" >}}) are cached separately from the
+    object they refer to.
+
+
+## Enabling CSC
+
+Use the [`CLIENT TRACKING`]({{< relref "/commands/client-tracking" >}})
+command to enable CSC from [`redis-cli`]({{< relref "/develop/connect/cli" >}}).
+You can also enable CSC when you connect to a server from one of the Redis
+[client libraries]({{< relref "/develop/connect/clients" >}}).
+
+## Usage recommendations
+
+Like any caching system, CSC has some limitations:
+
+-   The cache has only a limited amount of memory available. When the limit
+    is reached, the client must *evict* potentially useful items from the
+    cache to make room for new ones.
+-   Cache misses, tracking, and invalidation messages always add a slight
+    performance penalty.
+
+Below are a few guidelines to help you use CSC efficiently, within these
+limitations:
+
+-   **Use a separate connection for data that is not cache-friendly**:
+    Caching gives the most benefit
+    for keys that are read frequently and updated infrequently. However, you
+    may also have data such as counters and scoreboards that receive frequent
+    updates. In cases like this, the performance overhead of the invalidation
+    messages can be greater than the savings made by caching. Avoid this problem
+    by using a separate connection *without* CSC for any data that is
+    not cache-friendly.
+-   **Set a maximum time-to-live (TTL) for cached items**: The client libraries
```

Comment:
The TTL is not supported in this phase, so it has been removed recently. cc @uglide